### PR TITLE
fix: changes to accomodate files changes on edition (MSEDO-319)

### PR DESCRIPTION
**Pull request by sfdsilva**
MSEDO-319

### CODE REVIEW

File: src/main/java/uc/dei/mse/supportivecare/service/OutcomeService.java [hunk 1]
```diff
@@ -38,6 +46,22 @@ public OutcomeService(OutcomeRepository outcomeRepository, OutcomeMapper outcome
      */
     public OutcomeDTO save(OutcomeDTO outcomeDTO) {
         log.debug("Request to save Outcome : {}", outcomeDTO);
+        Set<DocumentDTO> outcomeDocumentsDTO = outcomeDTO.getDocuments();
```
Comment (DanielFran) [inline src/main/java/uc/dei/mse/supportivecare/service/OutcomeService.java:49]: remove

File: src/main/java/uc/dei/mse/supportivecare/service/OutcomeService.java [hunk 2]
```diff
@@ -38,6 +46,22 @@ public OutcomeService(OutcomeRepository outcomeRepository, OutcomeMapper outcome
...
+        Set<DocumentDTO> completedDocumentsDTO = new HashSet<>();
+
+        outcomeDocumentsDTO
```
Comment (DanielFran) [inline src/main/java/uc/dei/mse/supportivecare/service/OutcomeService.java:52]: Substitute outcomeDocumentsDTO by outcomeDTO.getDocuments()

File: src/main/java/uc/dei/mse/supportivecare/service/dto/OutcomeDTO.java [hunk 3]
```diff
@@ -65,6 +65,10 @@ public void setDocuments(Set<DocumentDTO> documents) {
         this.documents = documents;
     }
 
+    public void addToDocuments(Set<DocumentDTO> documents) {
```
Comment (DanielFran) [inline src/main/java/uc/dei/mse/supportivecare/service/dto/OutcomeDTO.java:68]: Not needed

File: src/main/java/uc/dei/mse/supportivecare/web/rest/OutcomeResource.java [hunk 4]
```diff
@@ -105,7 +105,10 @@ public OutcomeResource(
             throw new BadRequestAlertException("Invalid id", ENTITY_NAME, "idnull");
         }
 
-        outcomeDTO.setDocuments(documentContentMapper.multiPartFilesToDocuments(files));
+        if (!files.isEmpty()) {
+            outcomeDTO.addToDocuments(documentContentMapper.multiPartFilesToDocuments(files));
```
Comment (DanielFran) [inline src/main/java/uc/dei/mse/supportivecare/web/rest/OutcomeResource.java:109]: Substitute by outcomeDTO.getDocuments.addAll(documentContentMapper.multiPartFilesToDocuments(files));

**Review by DanielFran — CHANGES_REQUESTED** [review]

**Review by andrenns — APPROVED** [review]

**Pull request merged**